src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -70,6 +72,10 @@ public Renderer(IServiceProvider serviceProvider, ILoggerFactory loggerFactory)
 
             _serviceProvider = serviceProvider;
             _logger = loggerFactory.CreateLogger<Renderer>();
+
+            var componentActivator = serviceProvider.GetService<IComponentActivator>()
+                ?? DefaultComponentActivator.Instance;
```

Comment:
We can just register the default activator in DI.

Reply:
I did that at first, but it’s more breaking. People who use Renderer outside our default DI config would then get errors. Example: bUnit. 

Do you have reasons for saying it’s important to require it to be in DI?

Reply:
Could we change this so that there's two ctors for Renderer, one that takes the default vs one without? It's convenient we have the serviceprovider, but having it be an explicit dependency makes it much more obvious and avoids yucky service location

Reply:
Done